### PR TITLE
Auto-aim in combat mode at mobs on the floor is disabled after 10 tiles.

### PR DESCRIPTION
**Pull request by paganiy**
## About The Pull Request
As stated in the title, auto-aiming at prone targets is disabled after the projectile passed more than 10 tiles (the limit of vision of an ordinary spaceman is 9 tiles). Within 10 tiles, shooting in combat mode remains unchanged. This also does not apply to sniper rifles.
Here's a preview of how it works:

https://github.com/tgstation/tgstation/assets/126676387/d62fbb03-9f63-4a73-a32a-62b4d11e4515
## Why It's Good For The Game
The main problem of shooting in combat mode is shooting in busy corridors. Often, you are unwilling to become a participant in the security's pursuit of the antagonist, and receive unintended projectiles. You will be lucky if the projectile turns out to be non-lethal, but if it is a laser, you can get a second-degree burn or even a catastrophic one from just one shot. (depending on which laser was fired and where you were hit), which is not very pleasant. To stay unharmed and simplify the work of security, it is logical to lie down on the floor.
This will not revive the lie-down meta to dodge projectiles, because within sight of the spaceman and even a little further, lying targets will still be hit by projectiles. And if you decide to lie down during the chase to dodge the projectiles, then the pursuers will quickly catch up with you and shoot you in combat mode.
## Changelog
:cl:
balance: Auto-aim in combat mode at mobs on the floor is disabled after the projectile passes 10 tiles.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
I don't hate this concept, however I do sort of dislike the justification because I think bystanders getting hurt during shooting matches is both good and funny.

**Comment by paganiy** [comment]
> I don't hate this concept, however I do sort of dislike the justification because I think bystanders getting hurt during shooting matches is both good and funny.

If a laser didn't cause a third-degree burn from 1 shot when it hits an unprotected limb (a captain's laser can cause a catastrophic burn at all), it would be funny. But right now, It's just a waste of time for treatment., because someone is spamming lasers in the hallway without thinking about the consequences.

**Comment by DarkHunter7756** [comment]
> I don't hate this concept, however I do sort of dislike the justification because I think bystanders getting hurt during shooting matches is both good and funny.

Passersby will still take damage because they won't see the firefight 10 tiles away. And only after the first hit will they lie down

**Comment by SomeoneYouProbablyKnow** [comment]
> I don't hate this concept, however I do sort of dislike the justification because I think bystanders getting hurt during shooting matches is both good and funny.

Isn't people hearing laser fire and all dropping to the ground even funnier? Imagine like 4 people instantly resting and crawling to the nearest corner and one guy still gets hit with a stray shot because he didn't rest

**Comment by Jacquerel** [comment]
> > I don't hate this concept, however I do sort of dislike the justification because I think bystanders getting hurt during shooting matches is both good and funny.
> 
> Isn't people hearing laser fire and all dropping to the ground even funnier? Imagine like 4 people instantly resting and crawling to the nearest corner and one guy still gets hit with a stray shot because he didn't rest

yeah thats why i didnt close this or mark it do not merge or anything

**Comment by Pickle-Coding** [comment]
Shouldn't apply to sniper bullets. It would suck if someone decided to rest and then move perpendicular to the bullet trajectory, making it effectively impossible to hit them because you need to lead your aim, which you can't when you have to click their body.

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -621,13 +621,15 @@
 			return FALSE
 		if(HAS_TRAIT(living_target, TRAIT_IMMOBILIZED) && HAS_TRAIT(living_target, TRAIT_FLOORED) && HAS_TRAIT(living_target, TRAIT_HANDS_BLOCKED))
 			return FALSE
-		if(!hit_prone_targets)
+		if(hit_prone_targets)
 			var/mob/living/buckled_to = living_target.lowest_buckled_mob()
-			if(!buckled_to.density) // Will just be us if we're not buckled to another mob
-				return FALSE
-			if(living_target.body_position != LYING_DOWN)
+			if((istype(src, /obj/projectile/bullet/p50)) || (istype(src, /obj/projectile/bullet/pellet/stingball))) // doesn't apply to sniper rifles and ricocheted stingballs
```
Comment (paganiy) [inline code/modules/projectiles/projectile.dm:626]: I don't like this line. There might be a better option out there. However, I can't think of any alternative.

**Comment by moocowswag** [comment]
Targeting legs while firing should give a high chance of hitting prone people regardless of combat mode

File: code/modules/projectiles/projectile.dm (hunk 1 above)
Comment (Jacquerel) [reply code/modules/projectiles/projectile.dm:626]: Add a new var to bullets which describes whether they have this behaviour or not

Comment (paganiy) [reply code/modules/projectiles/projectile.dm]: thanks!

**Pull request merged**